kogitoq/kogito-common/src/main/java/life/genny/kogito/common/service/InitService.java

```diff
@@ -181,14 +215,66 @@ public void sendPCMs() {
 		KafkaUtils.writeMsg(KafkaTopic.WEBDATA, msg);
 	}
 
+	private Ask generateAddItemsAsk(String productCode, BaseEntity user) {
+		// find the question in the database
+		Question groupQuestion;
+		try {
+			groupQuestion = databaseUtils.findQuestionByCode(productCode, "QUE_ADD_ITEMS_GRP");
+		} catch (NoResultException e) {
+			throw new ItemNotFoundException("QUE_ADD_ITEMS_GRP", e);
+		}
+
+		Ask parentAsk = new Ask(groupQuestion);
+		parentAsk.setSourceCode(user.getCode());
+		parentAsk.setTargetCode(user.getCode());
+		parentAsk.setRealm(productCode);
+
+		List<EntityAttribute> capabilities = capMan.getEntityCapabilities(productCode, user);
+		
+		// Generate the Add Items asks from the capabilities
+		// Check if there is a def first
+		for(EntityAttribute capability : capabilities) {
```

Comment:
Can this loop be run in parallel, any interdependency?

Reply:
I don't think it is worth running the loop in parallel, since the number of items in the capabilities Collection will be small, the overhead of creating threads to run in a parallelStream will outweigh the benefits of having the parallelStream itself

Reply:
Makes sense.